plugins/pi4j-plugin-ffm/src/main/java/com/pi4j/plugin/ffm/providers/i2c/I2CDirect.java

```diff
@@ -0,0 +1,133 @@
+package com.pi4j.plugin.ffm.providers.i2c;
+
+import com.pi4j.context.Context;
+import com.pi4j.exception.InitializeException;
+import com.pi4j.io.i2c.I2C;
+import com.pi4j.io.i2c.I2CBase;
+import com.pi4j.io.i2c.I2CConfig;
+import com.pi4j.io.i2c.I2CProvider;
+import com.pi4j.plugin.ffm.common.file.FileDescriptorNative;
+import com.pi4j.plugin.ffm.common.ioctl.IoctlNative;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class I2CDirect extends I2CBase<I2CBusFFM> {
+    private static final Logger logger = LoggerFactory.getLogger(I2CDirect.class);
+    private static final FileDescriptorNative FILE = new FileDescriptorNative();
+    private static final IoctlNative IOCTL = new IoctlNative();
+
+    public I2CDirect(I2CProvider provider, I2CConfig config, I2CBusFFM i2CBus) {
+        super(provider, config, i2CBus);
+    }
+
+    @Override
+    public I2C initialize(Context context) throws InitializeException {
+        super.initialize(context);
+        return this;
+    }
+
+    @Override
+    public void close() {
+        super.close();
+        i2CBus.close();
+    }
+
+    /**
+     * Internal method.
+     * Writes the data array into the register address of device selected previously.
+     *
+     * @param data data array to be written
+     */
+    private int writeInternal(int register, byte[] data) {
+            i2CBus.selectDevice(config.device());
+            return i2CBus.execute(this, (i2cFileDescriptor) -> {
```

Comment:
Thread safety here everywhere... The execute method should perform the device selection on the bus and keep a lock on the bus